internal/v1/handler.go

```diff
@@ -498,6 +498,89 @@ func convertIgnoreImageTypeToSlice(ignoreImageTypes *[]ImageTypes) []string {
 	return stringSlice
 }
 
+func newComposeResponse(data []ComposesResponseItem, count int, first, last string) ComposesResponse {
+	return ComposesResponse{
+		Data: data,
+		Meta: struct {
+			Count int `json:"count"`
+		}{
+			count,
+		},
+		Links: struct {
+			First string `json:"first"`
+			Last  string `json:"last"`
+		}{first, last},
+	}
+}
```

Comment:
I would prefer two functions and types for meta and links so one could use struct literal to build the response but insert the metadata in easy way:

```
result := ComposesResponse{
  Data: // ...
  Meta: newMetaData(count, first, last),
  Links: newMetaLinks(count, first, last),
}
```

That seems more reusable than one function per response type.